android/src/main/java/com/expensify/reactnativekeycommand/KeyCommandModule.java

```diff
@@ -8,24 +8,136 @@
+        WritableMap params = new WritableNativeMap();
+        int action = keyEvent.getAction();
+        char pressedKey = (char) keyEvent.getUnicodeChar();
+
+        params.putString("input", String.valueOf(keyEvent.getDisplayLabel()).toLowerCase());
+
+        if (keyEvent.isCtrlPressed()) {
+            params.putInt("modifierFlags", KeyEvent.META_CTRL_MASK);
+        }
+        else if (keyEvent.isAltPressed()) {
+            params.putInt("modifierFlags", KeyEvent.META_ALT_MASK);
+        }
+        else if (keyEvent.isShiftPressed()) {
+            params.putInt("modifierFlags", KeyEvent.META_SHIFT_MASK);
+        } else {
+            params.putInt("modifierFlags", keyEvent.getModifiers());
+        }
+
+        return params;
+    }
+
+
     @Override
     @NonNull
     public String getName() {
         return NAME;
     }
 
-    // Example method
-    // See https://reactnative.dev/docs/native-modules-android
+    @Override
+    public Map<String, Object> getConstants() {
+        final Map<String, Object> constants = new HashMap<>();
+
+        constants.put("keyModifierShift", KeyEvent.META_SHIFT_MASK);
+        constants.put("keyModifierControl", KeyEvent.META_CTRL_MASK);
+        constants.put("keyModifierAlternate", KeyEvent.META_ALT_MASK);
+        constants.put("keyModifierCommand", KeyEvent.META_CTRL_MASK);
+        constants.put("keyModifierNumericPad", KeyEvent.KEYCODE_NUM);
+        constants.put("keyInputUpArrow", KeyEvent.KEYCODE_DPAD_UP);
+        constants.put("keyInputDownArrow", KeyEvent.KEYCODE_DPAD_DOWN);
+        constants.put("keyInputLeftArrow", KeyEvent.KEYCODE_DPAD_LEFT);
+        constants.put("keyInputRightArrow", KeyEvent.KEYCODE_DPAD_RIGHT);
+        constants.put("keyInputEscape", KeyEvent.KEYCODE_ESCAPE);
+
+        return constants;
+    }
+
+    @ReactMethod
+    public void registerKeyCommands(ReadableArray json, Promise promise) {
+        for (int i = 0; i < json.size(); i++) {
+            Set<Object> command = new HashSet<Object>();
+            command.add(json.getMap(i).getString("input"));
+            command.add(json.getMap(i).getInt("modifierFlags"));
+            commandsArray.add(command);
+        }
+
+        promise.resolve(null);
+    }
+
     @ReactMethod
-    public void multiply(double a, double b, Promise promise) {
-        promise.resolve(a * b);
+    public void unregisterKeyCommands(ReadableArray json, Promise promise) {
+        for (int ii = 0; ii < json.size(); ii++) {
+            Set<Object> command = new HashSet<Object>();
+            command.add(json.getMap(ii).getString("input"));
+            command.add(json.getMap(ii).getInt("modifierFlags"));
+            commandsArray.remove(command);
+        }
+
+        promise.resolve(null);
     }
+
+    @ReactMethod
+    public void addListener(String eventName) {}
+
+    @ReactMethod
+    public void removeListeners(Integer count) {}
```

Comment:
We are implementing the `NativeEventEmitter` interface for this module which require these methods to be exposed from nativeModule. Is there any other low level interface that be used instead of this bcz we don't need these methods. e.g. `EventEmitter`

Reply:
Added those two per RN Documentation, so I think we are good here until RN suppresses warnings here in the upcoming releases. 

https://reactnative.dev/docs/native-modules-android
https://github.com/facebook/flipper/issues/2707

Reply:
Ok, I still don't understand the use of these noop functions but it is fine as interface defines those. 

Are you planning to add turbo module implementation as well in future PRs?

Reply:
Haven’t looked at the implementation of turbo modules yet.